scheduler-server/scheduling-dsl-compiler/src/main.ts

```diff
@@ -21,10 +22,18 @@ const windowsAST = fs.readFileSync(
   `${process.env["SCHEDULING_DSL_COMPILER_ROOT"]}/src/libs/constraints/constraints-ast.ts`,
   "utf8"
 );
+const temporalPolyfillTypes = fs.readFileSync(
+    `${process.env["SCHEDULING_DSL_COMPILER_ROOT"]}/src/libs/TemporalPolyfillTypes.ts`,
+    'utf8',
+);
+const polyfills = fs.readFileSync(
```

Comment:
🔶 This `polyfills` variable seems unused - is it necessary?

Reply:
It should not be necessary if we are supplying the Temporal implementation via context as we do on line 89. The polyfills.ts looks to simply put Temporal on the globalThis and polyfill in Date.toTemporalInstant - which shouldn't be necessary as Date is generally not a great API - it was designed after the Java Date api that was subsequently deprecated and we shouldn't be passing any Date's to our stuff, so it shouldn't be necessary.

Reply:
Where we DO need it is actually imported into our process (`import 'polyfills.ts';` as the first line) so that we can correctly serialize/deserialize the Temporal types for transfer to/from the Java process.

Reply:
Command expansion has the correct (hopefully) working incantations for this, so you can reference what we have going on there https://github.com/NASA-AMMOS/aerie/blob/develop/command-expansion-server/src/worker.ts

Reply:
I think we'll likely need to borrow the Temporal serialization/deserialization code from the command expansion as well so that we can serialize everything correctly (something we DONT have to do on command expansion because we never have temporal objects left over once we have the seqJSON - which does the serialization within the toSeqJson call)

Reply:
It looks like they serialize/deserialize into a format that isn't necessarily what we want on the Java side of things, but we could also just roll with it and write a parser on the Java side to handle that format.

Reply:
Thanks for weighing in, Dylan :+1: It looks like this PR changes the serialization format to be an ISO8601 string, and uses `java.time.Duration.parse` to parse it. I'm not sure how it _becomes_ an ISO8601 string, maybe that's the default toJSON behavior...?

Reply:
Seems like the standard json stringification behavior